pkg/auth/requests/sar/sar.go

```diff
@@ -68,26 +78,49 @@ func (sar subjectAccessReview) checkUserCanImpersonateUser(sarClient v1.SubjectA
 	return result.Status.Allowed, nil
 }
 
-func (sar subjectAccessReview) checkUserCanImpersonateGroup(sarClient v1.SubjectAccessReviewInterface, user string, groups []string) (bool, error) {
-	for _, group := range groups {
-		review := authV1.SubjectAccessReview{
-			Spec: authV1.SubjectAccessReviewSpec{
-				User: user,
-				ResourceAttributes: &authV1.ResourceAttributes{
-					Verb:     "impersonate",
-					Resource: "groups",
-					Name:     group,
-				},
+func (sar subjectAccessReview) checkUserCanImpersonateGroup(sarClient v1.SubjectAccessReviewInterface, user string, group string) (bool, error) {
+	review := authV1.SubjectAccessReview{
+		Spec: authV1.SubjectAccessReviewSpec{
+			User: user,
+			ResourceAttributes: &authV1.ResourceAttributes{
+				Verb:     "impersonate",
+				Resource: "groups",
+				Name:     group,
 			},
-		}
+		},
+	}
 
-		result, err := sarClient.Create(context.TODO(), &review, metav1.CreateOptions{})
-		if err != nil {
-			return false, err
-		}
-		logrus.Debugf("Impersonate check result: %v", result)
-		if !result.Status.Allowed {
-			return false, nil
+	result, err := sarClient.Create(context.TODO(), &review, metav1.CreateOptions{})
+	if err != nil {
+		return false, err
+	}
+	logrus.Debugf("Impersonate check result: %v", result)
+
+	return result.Status.Allowed, nil
+}
+
+func (sar subjectAccessReview) checkUserCanImpersonateExtras(sarClient v1.SubjectAccessReviewInterface, user string, extras map[string][]string) (bool, error) {
+	for name, values := range extras {
+		for _, value := range values {
+			review := authV1.SubjectAccessReview{
+				Spec: authV1.SubjectAccessReviewSpec{
+					User: user,
+					ResourceAttributes: &authV1.ResourceAttributes{
+						Verb:     "impersonate",
+						Resource: "userextras/" + name,
+						Name:     value,
+					},
+				},
+			}
+
+			result, err := sarClient.Create(context.TODO(), &review, metav1.CreateOptions{})
```

Comment:
Q: Can we use `req.Context()` from `UserCanImpersonateExtras` instead of `context.TODO()` here?